src/utils/resize.js

```diff
@@ -0,0 +1,39 @@
+import { ref, onMounted, onUnmounted } from 'vue';
+
+const mobileBreakpoint = 1200;
+const isMobile = ref(window.innerWidth <= mobileBreakpoint);
+const resizeSubscribers = new Set();
+
+const onWindowResize = () => {
+  isMobile.value = window.innerWidth <= mobileBreakpoint;
+
+  // Notify all subscribed functions
+  resizeSubscribers.forEach((callback) => callback(isMobile.value));
+};
+
+let isInitialized = false;
+
+export function useResize() {
+  if (!isInitialized) {
+    onMounted(() => {
+      window.addEventListener('resize', onWindowResize);
+    });
+
+    onUnmounted(() => {
+      window.removeEventListener('resize', onWindowResize);
+    });
+
+    isInitialized = true;
+  }
+
+  const onResize = (callback) => {
+    resizeSubscribers.add(callback);
+    return () => resizeSubscribers.delete(callback); // Return an unsubscribe function
```

Comment:
I have a question here: When will this delete function be executed ? On a normal resize of the window, it is not executed, only the function definition is returned..

Reply:
here https://github.com/subugoe/tido/pull/540/files#diff-fc950e4a837261bc8651866090c3b578859d068322e29f41f88ed6fd0afbf63dR63

Reply:
what I see from the above link is 'watch(configStore.config, () => {
  updatePanelWidth(isMobile.value)
})'
Is this what you meant ?

Reply:
No, its in the PanelWrapper in the onUnount